modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -522,6 +582,28 @@ private void updateWatermark(Serializable value, Comparator comparator) {
     }
   }
 
+  private void updateWatermark(Serializable value, Comparator comparator) throws ObjectStoreException {
+    if (watermarkObjectStore.contains(WATERMARK_ITEM_OS_KEY)) {
+      Serializable currentValue = watermarkObjectStore.retrieve(WATERMARK_ITEM_OS_KEY);
+      if (compareWatermarks(currentValue, value, comparator) >= 0) {
+        return;
+      }
+      watermarkObjectStore.remove(WATERMARK_ITEM_OS_KEY);
+    }
+
+    updateRecentlyProcessedIds();
+    watermarkObjectStore.store(WATERMARK_ITEM_OS_KEY, value);
+  }
+
+  private void setCurrentWatermarkAsMinimumRejectWatermark(Serializable minimumRejectedByLimitPassingWatermark)
+      throws ObjectStoreException {
+    if (watermarkObjectStore.contains(WATERMARK_ITEM_OS_KEY)) {
+      watermarkObjectStore.retrieve(WATERMARK_ITEM_OS_KEY);
```

Comment:
Why retrieve if the value is not used? Did you leave this line by accident?

Reply:
My bad, removing it.